js/store.js

```diff
@@ -134,19 +256,74 @@ const store = {
     const firstParty = this.getWebsite(origin);
     const thirdParty = this.getWebsite(target);
 
+/*
+  if not {},
+
+  firstParty is expected to match this format:
+  {
+    thirdPartyHostnames: [
+      "www.thirdpartydomain.com",
+      ...
+    ]
+  }
+  thirdParty is expected to match this format:
+  {
+    firstPartyHostnames: [
+      "www.firstpartydomain.com",
+      ...
+    ],
+    isIgnored: true, (optional)
+    whiteListedFirstParty: "www.google.com" (optional)
+  }
+  */
+
+    let whiteListOverride = false;
+    let onWhiteList = false;
+
     if (!('thirdPartyHostnames' in firstParty)) {
       firstParty['thirdPartyHostnames'] = [];
     }
     if (!firstParty['thirdPartyHostnames'].includes(target)) {
-      firstParty['thirdPartyHostnames'].push(target);
-      newThirdParty = true;
+      // third party is not currently linked to first party
+      if (!thirdParty['isIgnored']) {
+        // third party needs a whitelist check
+        onWhiteList = this.onWhiteList(origin, target);
+        if (onWhiteList) {
+          // third party is whitelisted for this first party
+          // ignore the link for now
+          thirdParty['isIgnored'] = true;
+          thirdParty['whiteListedFirstParty'] = origin;
```

Comment:
Do you mean a third party might be whitelisted for multiple first parties?

I have to store this whitelisted first party because otherwise I lose which first party caused the third party to be ignored when a new third party request comes in.

In the case that the new third party request requires me to make visible the previously ignored third party, I need to know the original first party hostname that whitelisted the third party, so that I can update the third party's object (`firstPartyHostnames` key) and whitelisted first party's object (`thirdPartyHostnames` key) in `this_websites`.

I could make it an array instead perhaps?